File: solace/cmd/cooldown/Concentrate.java
package solace.cmd.cooldown;

import solace.game.*;

/**
 * Applies the "concentrate" buff that doubles the potency of cooldon actions
 * for ten seconds.
 * @author Ryan Sandor Richards
 */
public class Concentrate extends CooldownCommand {
  public Concentrate(Player p) {
    super("concentrate", p);
    setCooldownDuration(120);
  }

  public boolean execute(int level, Player target) {
    player.applyBuff("concentrating");
    return true;
  }
}

File: script/cooldowns/concentrate.js
'use strict';

/**
 * Applies the "concentrate" buff that doubles the potency of cooldon actions
 * for ten seconds.
 * @author Ryan Sandor Richards
 */
Commands.addCooldown('concentrate', {
  cooldownDuration: 180,
  run: function (player, target, level, cooldown) {
    player.applyBuff('concentrating');
    return true;
  }
});
